cpmpy/solvers/solver_interface.py

```diff
@@ -76,6 +76,7 @@ def __init__(self, name="dummy", cpm_model=None, subsolver=None):
         # initialise variable handling
         self.user_vars = set()  # variables in the original (non-transformed) model
         self._varmap = dict()  # maps cpmpy variables to native solver variables
```

Comment:
I guess this has to be a cpm_dict too? I don't think we store the varname, but the actual object as key... (or we should switch to the varname as key?)

Reply:
Yes, just not using the expression object as a key could also be a workaround, but that also involves computing the str everytime we use a set/dict, so not sure if it would be better